lib/api/CForecastRunner.cc

```diff
@@ -173,16 +173,21 @@ void CForecastRunner::forecastWorker() {
                     }
 
                     const TForecastModelWrapper& model = series.s_ToForecast.back();
-                    model_t::TDouble1VecDouble1VecPr support =
-                        model_t::support(model.s_Feature);
-                    bool success = model.s_ForecastModel->forecast(
-                        forecastJob.s_StartTime, forecastJob.forecastEnd(),
-                        forecastJob.s_BoundsPercentile, support.first, support.second,
+                    model_t::EFeature feature{model.s_Feature};
+                    core_t::TTime bucketLength{model.s_ForecastModel->params().bucketLength()};
+                    core_t::TTime startTime{model_t::sampleTime(
+                        feature, forecastJob.s_StartTime, bucketLength)};
+                    core_t::TTime endTime{model_t::sampleTime(
```

Comment:
The problem is this is feature specific. So it is tricky to push it higher up if the forecast is being run  over a job with multiple detectors with different features.

I could create a wrapper which implements the logic in the model library. I can't directly push the feature into the forecast function (because it is in the maths library which can't depend on EFeature). I could supply a call back to compute the offset start and end times and have this the wrapper from the model library.

Reply:
Alternatively, how about I add a function to actually run the forecast to model_t which wraps up this detail. Given we only have the maths::CTimeSeriesModel here (for good reason) this seems like it might be the cleanest option.

Reply:
ok, I see and agree that's to complicated.

What about inside of `model.s_ForecastModel->forecast(...)`?

Reply:
That hits the library dependency issue mentioned above. However, what about if I have a
`CForecastDataSink::SForecastModelWrapper::forecast` function which takes the forecast job. This could wrap all the functionality now in this loop?

Reply:
sounds good, I am also ok if we keep the current version given that alternatives are to complicated.

Reply:
I think I like the idea of wrapping this in SForecastModelWrapper. It seems more natural to me than in this loop which is really just about scheduling. I'll make it and see how it looks